x-pack/plugins/security_solution/common/utils/risk_score_modules.test.ts

```diff
@@ -0,0 +1,312 @@
+});
+
+describe('getRiskUserCreateMapScriptOptions', () => {
+  const options = getRiskUserCreateMapScriptOptions();
+  expect(options).toMatchSnapshot();
+});
+
+describe('getRiskUserCreateReduceScriptOptions', () => {
+  const options = getRiskUserCreateReduceScriptOptions();
+  expect(options).toMatchSnapshot();
+});
+
+describe('getRiskScoreIngestPipelineOptions', () => {
+  test('getRiskScoreIngestPipelineOptions - host', () => {
+    const options = getRiskScoreIngestPipelineOptions(RiskScoreEntity.host);
+    expect(options).toMatchSnapshot();
+  });
+
+  test('getRiskScoreIngestPipelineOptions - user', () => {
+    const options = getRiskScoreIngestPipelineOptions(RiskScoreEntity.user);
+    expect(options).toMatchSnapshot();
+  });
+});
+
+describe('getCreateRiskScoreIndicesOptions', () => {
+  test('getCreateRiskScoreIndicesOptions - host', () => {
+    const options = getCreateRiskScoreIndicesOptions({
+      spaceId: mockSpaceId,
+      riskScoreEntity: RiskScoreEntity.host,
+    });
+    expect(options).toMatchSnapshot();
+  });
+
+  test('getCreateRiskScoreIndicesOptions - user', () => {
+    const options = getCreateRiskScoreIndicesOptions({
+      spaceId: mockSpaceId,
+      riskScoreEntity: RiskScoreEntity.user,
+    });
+    expect(options).toMatchSnapshot();
+  });
+});
+
+describe('getCreateRiskScoreLatestIndicesOptions', () => {
+  test('getCreateRiskScoreLatestIndicesOptions - host', () => {
+    const options = getCreateRiskScoreLatestIndicesOptions({
+      spaceId: mockSpaceId,
+      riskScoreEntity: RiskScoreEntity.host,
+    });
+    expect(options).toMatchSnapshot();
+  });
+
+  test('getCreateRiskScoreLatestIndicesOptions - user', () => {
+    const options = getCreateRiskScoreLatestIndicesOptions({
+      spaceId: mockSpaceId,
+      riskScoreEntity: RiskScoreEntity.user,
+    });
+    expect(options).toMatchSnapshot();
+  });
+});
+
+describe('getCreateMLHostPivotTransformOptions', () => {
+  test('getCreateMLHostPivotTransformOptions', () => {
+    const options = getCreateMLHostPivotTransformOptions({
+      spaceId: mockSpaceId,
+    });
+    expect(options).toMatchSnapshot();
+  });
+});
+
+describe('getCreateMLUserPivotTransformOptions', () => {
+  test('getCreateMLUserPivotTransformOptions', () => {
+    const options = getCreateMLUserPivotTransformOptions({
+      spaceId: mockSpaceId,
+    });
+    expect(options).toMatchSnapshot();
+  });
+});
+
+describe('getCreateLatestTransformOptions', () => {
+  test('getCreateLatestTransformOptions - host', () => {
+    const options = getCreateLatestTransformOptions({
+      spaceId: mockSpaceId,
+      riskScoreEntity: RiskScoreEntity.host,
+    });
+    expect(options).toMatchSnapshot();
+  });
+
+  test('getCreateLatestTransformOptions - user', () => {
+    const options = getCreateLatestTransformOptions({
+      spaceId: mockSpaceId,
+      riskScoreEntity: RiskScoreEntity.user,
+    });
+    expect(options).toMatchSnapshot();
+  });
+});
```

Comment:
describe('Risk Score Modules', () => {
  [RiskScoreEntity.host, RiskScoreEntity.user].forEach((entity) => {
    test(`getRiskScorePivotTransformId - ${entity}`, () => {
      const id = getRiskScorePivotTransformId(entity, mockSpaceId);
      expect(id).toMatchInlineSnapshot(`"ml_${entity}riskscore_pivot_transform_customSpaceId"`);
    });
    test(`getIngestPipelineName - ${entity}`, () => {
      const name = getIngestPipelineName(entity);
      expect(name).toMatchInlineSnapshot(`"ml_${entity}riskscore_ingest_pipeline_default"`);
    });
    test(`getPivotTransformIndex - ${entity}`, () => {
      const index = getPivotTransformIndex(entity, mockSpaceId);
      expect(index).toMatchInlineSnapshot(`"ml_${entity}_risk_score_customSpaceId"`);
    });
    test(`getLatestTransformIndex - ${entity}`, () => {
      const index = getLatestTransformIndex(entity, mockSpaceId);
      expect(index).toMatchInlineSnapshot(`"ml_${entity}_risk_score_latest_customSpaceId"`);
    });
    test(`getRiskScoreLevelScriptId - ${entity}`, () => {
      const index = getRiskScoreLevelScriptId(entity);
      expect(index).toMatchInlineSnapshot(`"ml_${entity}riskscore_levels_script_default"`);
    });
    test(`getRiskScoreInitScriptId - ${entity}`, () => {
      const index = getRiskScoreInitScriptId(entity);
      expect(index).toMatchInlineSnapshot(`"ml_${entity}riskscore_init_script_default"`);
    });
    test(`getRiskScoreMapScriptId - ${entity}`, () => {
      const index = getRiskScoreMapScriptId(entity);
      expect(index).toMatchInlineSnapshot(`"ml_${entity}riskscore_map_script_default"`);
    });
    test(`getRiskScoreReduceScriptId - ${entity}`, () => {
      const index = getRiskScoreReduceScriptId(entity);
      expect(index).toMatchInlineSnapshot(`"ml_${entity}riskscore_reduce_script_default"`);
    });
    test(`getLegacyIngestPipelineName - ${entity}`, () => {
      const name = getLegacyIngestPipelineName(entity);
      expect(name).toMatchInlineSnapshot(`"ml_${entity}riskscore_ingest_pipeline"`);
    });
    test(`getLegacyRiskScoreLevelScriptId - ${entity}`, () => {
      const index = getLegacyRiskScoreLevelScriptId(entity);
      expect(index).toMatchInlineSnapshot(`"ml_${entity}riskscore_levels_script"`);
    });
    test(`getLegacyRiskScoreInitScriptId - ${entity}`, () => {
      const index = getLegacyRiskScoreInitScriptId(entity);
      expect(index).toMatchInlineSnapshot(`"ml_${entity}riskscore_init_script"`);
    });
    test(`getLegacyRiskScoreMapScriptId - ${entity}`, () => {
      const index = getLegacyRiskScoreMapScriptId(entity);
      expect(index).toMatchInlineSnapshot(`"ml_${entity}riskscore_map_script"`);
    });
    test(`getLegacyRiskScoreReduceScriptId - ${entity}`, () => {
      const index = getLegacyRiskScoreReduceScriptId(entity);
      expect(index).toMatchInlineSnapshot(`"ml_${entity}riskscore_reduce_script"`);
    });
    test(`getRiskScoreIngestPipelineOptions - ${entity}`, () => {
      const options = getRiskScoreIngestPipelineOptions(entity);
      expect(options).toMatchSnapshot();
    });
    test(`getCreateRiskScoreIndicesOptions - ${entity}`, () => {
      const options = getCreateRiskScoreIndicesOptions({
        spaceId: mockSpaceId,
        riskScoreEntity: entity,
      });
      expect(options).toMatchSnapshot();
    });
    test(`getCreateRiskScoreLatestIndicesOptions - ${entity}`, () => {
      const options = getCreateRiskScoreLatestIndicesOptions({
        spaceId: mockSpaceId,
        riskScoreEntity: entity,
      });
      expect(options).toMatchSnapshot();
    });
    test(`getCreateLatestTransformOptions - ${entity}`, () => {
      const options = getCreateLatestTransformOptions({
        spaceId: mockSpaceId,
        riskScoreEntity: entity,
      });
      expect(options).toMatchSnapshot();
    });
    test(`getCreateML${
      entity.charAt(0).toUpperCase() + entity.slice(1)
    }PivotTransformOptions`, () => {
      const fn =
        entity === RiskScoreEntity.host
          ? getCreateMLHostPivotTransformOptions
          : getCreateMLUserPivotTransformOptions;
      const options = fn({
        spaceId: mockSpaceId,
      });
      expect(options).toMatchSnapshot();
    });

    test(`getRisk${
      entity.charAt(0).toUpperCase() + entity.slice(1)
    }CreateLevelScriptOptions`, () => {
      const fn =
        entity === RiskScoreEntity.host
          ? getRiskHostCreateLevelScriptOptions
          : getRiskUserCreateLevelScriptOptions;
      const options = fn();
      expect(options).toMatchSnapshot();
    });

    test(`getRisk${
      entity.charAt(0).toUpperCase() + entity.slice(1)
    }CreateInitScriptOptions`, () => {
      const fn =
        entity === RiskScoreEntity.host
          ? getRiskHostCreateInitScriptOptions
          : // @angela is there not a user function for this?
            getRiskUserCreateInitScriptOptions;
      const options = fn();
      expect(options).toMatchSnapshot();
    });

    test(`getRisk${entity.charAt(0).toUpperCase() + entity.slice(1)}CreateMapScriptOptions`, () => {
      const fn =
        entity === RiskScoreEntity.host
          ? getRiskHostCreateMapScriptOptions
          : getRiskUserCreateMapScriptOptions;
      const options = fn();
      expect(options).toMatchSnapshot();
    });

    test(`getRisk${
      entity.charAt(0).toUpperCase() + entity.slice(1)
    }CreateReduceScriptOptions`, () => {
      const fn =
        entity === RiskScoreEntity.host
          ? getRiskHostCreateReduceScriptOptions
          : getRiskUserCreateReduceScriptOptions;
      const options = fn();
      expect(options).toMatchSnapshot();
    });
  });
});